### PR TITLE
[4.0] com_privacy consents

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
Updates the tmpl for the consents view to match J4 standards and styles
Add name column
Add search by name filter
Rename "Created" column header to "Consented"

### Testing Instructions
To test it will be easier if you have some data. These three rows should cover all the options - obviously replace jos_ with your own db prefix AND user_id with a valid id
```
INSERT INTO `jos_privacy_consents` (`id`, `user_id`, `state`, `created`, `subject`, `body`, `remind`, `token`) VALUES (1, 540, 1, '2019-07-01 19:36:00', 'title', 'body', 0, '');
INSERT INTO `jos_privacy_consents` (`id`, `user_id`, `state`, `created`, `subject`, `body`, `remind`, `token`) VALUES (2, 540, 0, '2018-06-01 12:01:00', 'title', 'body', 0, '');
INSERT INTO `jos_privacy_consents` (`id`, `user_id`, `state`, `created`, `subject`, `body`, `remind`, `token`) VALUES (3, 540, -1, '2019-07-01 07:16:00', 'title', 'body', 0, '');

```

### Expected result
image

Other parts of com_privacy are beyond the scope of this PR

### CODE REVIEW

**Comment by 1apweb** [comment]
I have tested this item :white_check_mark: successfully on 2c7508ddab3c8c53828ca251b7c9b350c1383510<br /><br />Hi, everything seems ok. But the drop-down menu Filter Option does not close for choices Status and Subjects.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25744">issues.joomla.org/tracker/joomla-cms/25744</a>.</sub>

File: administrator/components/com_privacy/Model/ConsentsModel.php [hunk 1]
```diff
@@ -80,6 +82,10 @@ protected function getListQuery()
 			{
 				$query->where($db->quoteName('a.user_id') . ' = ' . (int) substr($search, 4));
 			}
+			elseif (stripos($search, 'name:') === 0)
+			{
+				$query->where($db->quoteName('u.name') . ' LIKE ' . (int) substr($search, 5));
```
Comment (Quy) [inline administrator/components/com_privacy/Model/ConsentsModel.php]: Should be similar to lines 91-92.

Comment (brianteeman) [reply administrator/components/com_privacy/Model/ConsentsModel.php]: sorry dont understand what you mean

Comment (brianteeman) [reply administrator/components/com_privacy/Model/ConsentsModel.php]: got it now and fixed - thanks

File: administrator/components/com_privacy/Model/ConsentsModel.php [hunk 2]
```diff
@@ -80,6 +82,11 @@ protected function getListQuery()
 			{
 				$query->where($db->quoteName('a.user_id') . ' = ' . (int) substr($search, 4));
 			}
+			elseif (stripos($search, 'name:') === 0)
+			{
+				$search = $db->quote('%' . $db->escape($search, true) . '%');
```
Comment (Quy) [inline administrator/components/com_privacy/Model/ConsentsModel.php]:
```suggestion
				$search = $db->quote('%' . $db->escape(substr($search, 5), true) . '%');
```

File: administrator/components/com_privacy/Model/ConsentsModel.php [hunk 3]
```diff
@@ -80,6 +82,11 @@ protected function getListQuery()
...
+				$query->where($db->quoteName('u.name') . ' LIKE ' . (int) substr($search, 5));
```
Comment (Quy) [inline administrator/components/com_privacy/Model/ConsentsModel.php]:
```suggestion
				$query->where($db->quoteName('u.name') . ' LIKE ' . $search);
```

Comment (richard67) [reply administrator/components/com_privacy/Model/ConsentsModel.php]: Quy is right, this here HAS to be changed.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 8147e99ee79b301236681fd03978882660998861<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25744">issues.joomla.org/tracker/joomla-cms/25744</a>.</sub>

**Comment by richard67** [comment]
> Hi, everything seems ok. But the drop-down menu Filter Option does not close for choices Status and Subjects.

I can confirm @1apweb ' s finding and can confirm that it happens with and without this PR in the same way.

**Comment by richard67** [comment]
@brianteeman Is @1apweb ' s finding about filter drop down menu not closing worth a new issue? If so, will you write one or shall I do?

**Comment by brianteeman** [comment]
It's an old issue addressed by new template

**Comment by richard67** [comment]
@1apweb Could you test again? There have been made changes since you've tested.

**Comment by Quy** [comment]
Under Filter Options dropdown:
- change Created ascending/descending to Consented ascending/descending
- add Name ascending/descending

**Comment by Quy** [comment]
Does #24188 apply here?

**Comment by brianteeman** [comment]
@quy I hope not ;)  From my reading of that it was only until @wilsonge had merged 3.9.0 - and he has

**Comment by brianteeman** [comment]
Changes requested by @quy have been applied

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on eb90f8b66ce4b1b5154c594283afe7d5800883c7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25744">issues.joomla.org/tracker/joomla-cms/25744</a>.</sub>

**Comment by ghost** [comment]
@1apweb can you please retest?

**Comment by wilsonge** [comment]
> @Quy I hope not ;) From my reading of that it was only until @wilsonge had merged 3.9.0 - and he has

Correct. That issue is just open still so i can track my cleanup pile from the merges.

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on eb90f8b66ce4b1b5154c594283afe7d5800883c7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25744">issues.joomla.org/tracker/joomla-cms/25744</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**